File: CS/E3024.Web/ReportResolver.cs
using System.ComponentModel.Composition;
using System.Web;
using DevExpress.XtraReports.Service.Extensions;
using DevExpress.XtraReports.UI;

namespace E3024.Web {
    [Export(typeof(IReportResolver))]
    public class ReportResolver : IReportResolver {
        #region IReportResolver Members

        XtraReport IReportResolver.Resolve(string reportName, bool getParameters) {
            string layoutFile = GetReportPath(reportName);
            return XtraReport.FromFile(layoutFile, true);
        }

        #endregion

        static string GetReportPath(string reportName) {
            return HttpContext.Current.Server.MapPath("~/App_Data/" + reportName + ".repx");
        }
    }
}

File: VB/E3024.Web/ReportResolver.vb
Imports System.ComponentModel.Composition
Imports System.Web
Imports DevExpress.XtraReports.Service.Extensions
Imports DevExpress.XtraReports.UI

<Export(GetType(IReportResolver))>
Public Class ReportResolver
    Implements IReportResolver
    Public Function Resolve(reportName As String, getParameters As Boolean) As XtraReport Implements IReportResolver.Resolve
        Dim layoutFile = GetReportPath(reportName)
        Return XtraReport.FromFile(layoutFile, True)
    End Function

    Shared Function GetReportPath(reportName As String) As String
        Return HttpContext.Current.Server.MapPath("~/App_Data/" & reportName & ".repx")
    End Function
End Class
